lib/rules/no-throw-literal.js

```diff
@@ -1,12 +1,32 @@
 /**
  * @fileoverview Rule to restrict what can be thrown as an exception.
  * @author Dieter Oberkofler
+ * @copyright 2015 Ian VanSchooten. All rights reserved.
  * @copyright 2015 Dieter Oberkofler. All rights reserved.
  */
 
 "use strict";
 
 //------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Determine if a node is considered literal (including concatenation)
+ * @param  {ASTNode}  node  ASTNode to check for only literals
+ * @returns {Boolean}       True if considered literal, false otherwise
+ */
+function isLiteral(node) {
+    if (node.type === "Literal") {
+        return true;
+    }
+
+    if (node.type === "BinaryExpression") {
+        return true;
```

Comment:
This is where we start to fall down the rabbit hole. Every rule has gaps in coverage because we can't possibly account for every possibility through static analysis. With this, we are starting to cross the line over static analysis into runtime interpretation, which starts to get dangerous because that's when people start asking questions like, "how come it doesn't flag throw foo() because foo() returns a string?"

The original intent of the rule was to flag `throw "foo"`, we expanded to other literals because it was cheap. I'm on the fence as to whether checking for binary expressions makes sense here because, as Ilya stated, that's not what the rule's name implies.

As for changing the name, it's too late to do that now. Even though 1.0.0 isn't final, I don't want to start changing rule names at this stage.

I think the fundamental question here is if flagging binary expressions in this situation makes sense? If yes, then we can answer whether it belongs in this rule or not.